src/mon/ConfigMonitor.cc

```diff
@@ -521,11 +521,9 @@ bool ConfigMonitor::prepare_command(MonOpRequestRef op)
     }
     goto update;
   } else if (prefix == "config reset") {
-    int64_t num = -1;
-    cmd_getval(g_ceph_context, cmdmap, "num", num);
-    int64_t revert_to = num;
-    if (revert_to < 0) // revert to last version
-      revert_to = version - 1;
+    int64_t revert_to = -1;
+    cmd_getval(g_ceph_context, cmdmap, "num", revert_to);
+    ceph_assert(revert_to >= 0);
```

Comment:
this shoudl return EINVAL, not assert, right?

the original mimic code looks like
```
    if (num < 0 ||
	(num > 0 && num > (int64_t)version)) {
      err = -EINVAL;
      ss << "must specify a valid version to revert to";
      goto reply;
    }
```

Reply:
[root@ /clove/xxG/ceph/build]# bin/ceph config reset
Invalid command: missing required parameter num(<int[0-]>)
config reset <int[0-]> :  Revert configuration to a historical version specified by <num>
Error EINVAL: invalid command
[root@ /clove/xxG/ceph/build]# bin/ceph config reset -1
Invalid command: -1 not in range [0]
config reset <int[0-]> :  Revert configuration to a historical version specified by <num>
Error EINVAL: invalid command

Reply:
@liewegas The `num ` is mandatory and the range checker 'range|0' will automatically prevent user passing in any negative value.

Reply:
we should still return EINVAL.  a librados user or mgr module can send arbitrary JSON

Reply:
@liewegas okay, fixed.